src/hooks/use-sorting.ts

```diff
@@ -0,0 +1,67 @@
+import type { HyperCube, Model } from "../types/QIX";
+import type { ChangeActivelySortedHeader, ChangeSortOrder, Header, SortDirection } from "../types/types";
+
+interface UseSortingApi {
+  changeSortOrder: ChangeSortOrder;
+  changeActivelySortedHeader: ChangeActivelySortedHeader;
+}
+
+interface UseSorting {
+  (model: Model, hyperCube: HyperCube): UseSortingApi;
+}
+
+const useSorting: UseSorting = (model, qHyperCube) => {
```

Comment:
This is  not a hook as it's not bound by the rules of hooks.

You should however make it a hook by wrapping the creation of `api` in a `useMemo`.